iceoryx_posh/include/iceoryx_posh/popo/rpc_header.hpp

```diff
@@ -0,0 +1,160 @@
+
+#ifndef IOX_POSH_POPO_RPC_HEADER_HPP
+#define IOX_POSH_POPO_RPC_HEADER_HPP
+
+#include "iceoryx_posh/iceoryx_posh_types.hpp"
+#include "iceoryx_posh/mepoo/chunk_header.hpp"
+
+#include <cstdint>
+
+namespace iox
+{
+namespace popo
+{
+class RpcBaseHeader
+{
+  public:
+    /// @brief Constructs and initializes a RpcBaseHeader
+    /// @param[in] clientQueueUniquePortId is the UniquePortId of the port owning the client queue where the response
+    /// shall be delivered
+    /// @param[in] lastKnownClientQueueIndex is the last know index of the client queue in the ChunkDistributor for fast
+    /// lookup
+    /// @param[in] sequenceId is a custom ID to map a response to a request
+    explicit RpcBaseHeader(const UniquePortId& clientQueueUniquePortId,
+                           const uint32_t lastKnownClientQueueIndex,
+                           const int64_t sequenceId,
+                           const uint8_t rpcHeaderVersion);
+
+    RpcBaseHeader(const RpcBaseHeader& other) = delete;
+    RpcBaseHeader& operator=(const RpcBaseHeader&) = delete;
+    RpcBaseHeader(RpcBaseHeader&& rhs) = default;
+    RpcBaseHeader& operator=(RpcBaseHeader&& rhs) = default;
+    ~RpcBaseHeader() = default;
+
+    /// @brief From the 2.0 release onward, this must be incremented for each incompatible change, e.g.
+    ///            - data width of members changes
+    ///            - members are rearranged
+    ///            - semantic meaning of a member changes
+    ///        in any of RpcBaseHeader, RequestHeader or ResponseHeader!
+    static constexpr uint8_t RPC_HEADER_VERSION{1U};
+
+    static constexpr uint32_t UNKNOWN_CLIENT_QUEUE_INDEX{std::numeric_limits<uint32_t>::max()};
+
+    /// @brief The RpcBaseHeader version is used to detect incompatibilities for record&replay functionality
+    /// @return the RpcBaseHeader version
+    uint8_t getRpcHeaderVersion() const;
+
+    /// @briet Obtains the sequence ID of the RPC message
+    /// @return the sequenceId of the RPC message
+    int64_t getSequenceId() const noexcept;
+
+    /// @brief Get the pointer to the ChunkHeader
+    /// @return the pointer to the ChunkHeader
+    mepoo::ChunkHeader* getChunkHeader() noexcept;
+
+    /// @brief Get the const pointer to the ChunkHeader
+    /// @return the const pointer to the ChunkHeader
+    const mepoo::ChunkHeader* getChunkHeader() const noexcept;
+
+    /// @brief Get the pointer to the user-payload
+    /// @return the pointer to the user-payload
+    void* getUserPayload() noexcept;
+
+    /// @brief Get the const pointer to the user-payload
+    /// @return the const pointer to the user-payload
+    const void* getUserPayload() const noexcept;
+
+    friend class ServerPortUser;
+
+  protected:
+    uint8_t m_rpcHeaderVersion{RPC_HEADER_VERSION};
+    uint32_t m_lastKnownClientQueueIndex{UNKNOWN_CLIENT_QUEUE_INDEX};
+    UniquePortId m_clientQueueUniquePortId;
+    int64_t m_sequenceId{0};
+};
+
+class RequestHeader : public RpcBaseHeader
```

Comment:
I know we have the rule of one class per file, but these are closely related and therefore I ask for an exception of this rule for these header

Reply:
Your exception will be granted if you grant me one favor which I may ask for in the future.